### PR TITLE
wr-293: achievements redesign

### PR DESCRIPTION
**Pull request by artemmatiushenko1**
image
image

### CODE REVIEW

File: frontend/src/pages/profile/components/achievement/achievement.tsx [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
+  const achievementEntityClassName: string | undefined = {
+    'comments': styles.comments,
+    'articles': styles.articles,
+  }[referenceTable];
+
+  const AchievementImage: React.FC<React.SVGProps<SVGSVGElement>> | undefined =
+    {
+      'comments': CommentsAchievementIcon,
+      'articles': ArticlesAchievementIcon,
+    }[referenceTable];
```
Comment (andanf-e) [inline frontend/src/pages/profile/components/achievement/achievement.tsx]: Let's move it out to separate component

Comment (andanf-e) [reply frontend/src/pages/profile/components/achievement/achievement.tsx]: And pass it as Icon if possible

File: frontend/src/pages/profile/components/achievement/achievement.tsx [hunk 2]
```diff
@@ -1,29 +1,62 @@
...
+            cx="50"
+            cy="50"
+            className={styles.progress}
+            r={AchievementItemConfig.PROGRESS_RADIUS}
+            strokeDasharray={getRadialProgressStyleString(
+              progress,
+              AchievementItemConfig.PROGRESS_RADIUS,
+            )}
+          />
+        </svg>
```
Comment (andanf-e) [inline frontend/src/pages/profile/components/achievement/achievement.tsx:48]: Can we add progress colors for this circle?
Like
Green - for `100%` **|** Orange\Yellow - for  `progress > 30%` **|** Gray - for `progress < 30%`

File: frontend/src/pages/profile/components/achievement/achievement.tsx [hunk 3]
```diff
@@ -11,19 +10,47 @@ type Properties = {
   className?: string;
 };
 
-const Achievement: FC<Properties> = ({ achievement, className }) => {
-  const { name, progress } = achievement;
-  const statusStyleClass = getProgressStyleClass(progress);
+const Achievement: React.FC<Properties> = ({ achievement }) => {
+  const { name, referenceTable, progress } = achievement;
+  const radius = 45;
+  const circumference = Math.PI * 2 * radius;
```
Comment (PablloZz) [inline frontend/src/pages/profile/components/achievement/achievement.tsx]: I guess it should be RADIUS as it said in the B22 rule

**Review by andanf-e — APPROVED** [review]

**Pull request merged**